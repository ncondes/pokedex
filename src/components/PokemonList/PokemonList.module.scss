@use '@/scss/mixins' as *;

.pokemon_grid {
   width: 100%;
   display: grid;
   place-items: center;
   grid-template-columns: repeat(1, minmax(280px, 1fr));
   gap: 2.5rem;

   @include breakpoint-up('md') {
      grid-template-columns: repeat(2, minmax(280px, 1fr));
   }

   @include breakpoint-up('lg') {
      grid-template-columns: repeat(3, minmax(280px, 1fr));
   }

   @include breakpoint-up('xl') {
      grid-template-columns: repeat(4, minmax(280px, 1fr));
   }
}

.no_results {
   font-size: 2rem;
   font-weight: 600;
   text-align: center;
}
