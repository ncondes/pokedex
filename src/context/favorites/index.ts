export { FavoritesContext } from './FavoritesContext'
export { FavoritesProvider } from './FavoritesProvider'
