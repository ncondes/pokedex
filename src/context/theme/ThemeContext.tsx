'use client'

import { createContext } from 'react'

interface Props {
   theme: 'light' | 'dark'
   toggleTheme: () => void
   useTheme: (className: string) => string
}

export const ThemeContext = createContext({} as Props)
