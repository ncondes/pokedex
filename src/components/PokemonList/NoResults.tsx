'use client'
import { useContext } from 'react'
import styles from './PokemonList.module.scss'
import { ThemeContext } from '@/context/theme'

export const NoResults = () => {
   const { useTheme } = useContext(ThemeContext)

   return (
      <p className={`${styles.no_results} ${useTheme('font')}`}>
         No Pokemons Found.
      </p>
   )
}
