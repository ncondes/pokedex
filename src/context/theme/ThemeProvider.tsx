'use client'

import { FC, PropsWithChildren, useState } from 'react'
import { ThemeContext } from './ThemeContext'
import { getLocalTheme, setLocalTheme } from '@/utils'

const INITIAL_THEME = getLocalTheme() as 'light' | 'dark'

export const ThemeProvider: FC<PropsWithChildren> = ({ children }) => {
   const [theme, setTheme] = useState<'light' | 'dark'>(INITIAL_THEME)

   const toggleTheme = () => {
      const t = theme === 'light' ? 'dark' : 'light'
      setTheme(t)
      setLocalTheme(t)
   }

   const useTheme = (className: string): string => `${theme}-${className}`

   return (
      <ThemeContext.Provider value={{ theme, toggleTheme, useTheme }}>
         {children}
      </ThemeContext.Provider>
   )
}
