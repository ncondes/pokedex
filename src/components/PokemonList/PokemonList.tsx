import { FC } from 'react'
import { Pokemon } from '@/interfaces'
import { NoResults } from './NoResults'
import styles from './PokemonList.module.scss'
import { PokemonItem } from '../PokemonItem'

interface Props {
   pokemons: Pokemon[]
}

export const PokemonList: FC<Props> = ({ pokemons }) => {
   return (
      <>
         {pokemons.length === 0 ? (
            <NoResults />
         ) : (
            <div className={styles.pokemon_grid}>
               {pokemons.map((pokemon) => (
                  <PokemonItem key={pokemon.id} pokemon={pokemon} />
               ))}
            </div>
         )}
      </>
   )
}
