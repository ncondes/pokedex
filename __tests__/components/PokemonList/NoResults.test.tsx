import { NoResults } from '@/components/PokemonList/NoResults'
import { ThemeProvider } from '@/context/theme'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

describe('NoResults', () => {
   it('renders the correct text', () => {
      const { getByText } = render(
         <ThemeProvider>
            <NoResults />
         </ThemeProvider>
      )

      const noResults = getByText('No Pokemons Found.')

      expect(noResults).toBeInTheDocument()
   })

   it('should match the snapshot', () => {
      const { container } = render(
         <ThemeProvider>
            <NoResults />
         </ThemeProvider>
      )

      expect(container.firstChild).toMatchSnapshot()
   })
})
