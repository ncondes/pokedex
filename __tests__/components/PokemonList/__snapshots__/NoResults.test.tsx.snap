// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResults should match the snapshot 1`] = `
<p
  class="no_results light-font"
>
  No Pokemons Found.
</p>
`;
